client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/impl/channels/DefaultOperationTcpChannel.java

```diff
@@ -116,8 +116,13 @@ public void run() {
         @Override
         public void onMessage(ConnAck message) {
             LOG.info("ConnAck ({}) message received for channel [{}]", message.getReturnCode(), getId());
+
             if (message.getReturnCode() != ReturnCode.ACCEPTED) {
                 LOG.error("Connection for channel [{}] was rejected: {}", getId(), message.getReturnCode());
+                if (message.getReturnCode() == ReturnCode.REFUSE_BAD_CREDENTIALS) {
```

Comment:
This type of error is not specific to TCP channel, and should also be handled in HTTP channel